opentelemetry/src/context.rs

```diff
@@ -381,17 +388,115 @@ impl Hasher for IdHasher {
     }
 }
 
+/// A stack for keeping track of the [`Context`] instances that have been attached
+/// to a thread.
+///
+/// The stack allows for popping of contexts by position, which is used to do out
+/// of order dropping of [`ContextGuard`] instances. Only when the top of the
+/// stack is popped, the topmost [`Context`] is actually restored.
+///
+/// The stack relies on the fact that it is thread local and that the
+/// [`ContextGuard`] instances that are constructed using it can't be shared with
+/// other threads.
```

Comment:
Nit, I think "can't be shared with other threads" written like this usually means `!Sync` while here the guards are just `!Send`.

I think changing it to "can't be moved to other threads" would be clearer.